main.py

```diff
@@ -0,0 +1,169 @@
+    cursor.rowfactory = lambda *args: dict(
+        zip([d[0] for d in cursor.description], args)
+    )
+    rows = cursor.fetchall()
+    conn.close()
+
+    logging.info(f"{len(rows)} records in oracle database received.")
+    return rows
+
+
+def generate_record_id(call_number, timestamp):
+    """
+    the records from the oracle database do not have record ids associated with them
+    the coldfusion app added IDs generated by Coldfusion using #hash(call_number,"SHA")#
+    this function does the same using hashlib and appending our timestamp as before
+    :param call_number: call_number
+    :param timestamp:
+    :return: traffic_incident_id string
+    """
+    # create coldfusion hash id
+    cf_id = hashlib.sha1(str.encode(call_number)).hexdigest().upper()
+    #  compose record id from entry identifier (which is not wholly unique) and publication timestamp
+    return "{}_{}".format(cf_id, timestamp)
+
+
+def get_active_records():
+    """
+    Query active records from postgrest endpoint
+    :return: list of active records (dict)
+    """
+    active_records_endpoint = f"{PGREST_ENDPOINT}?traffic_report_status=eq.ACTIVE"
+
+    active_records_response = requests.get(active_records_endpoint, headers=headers)
+    active_records_response.raise_for_status()
+    return active_records_response.json()
+
+
+def format_record(incident):
+    """
+    Formats data from incident into record dict to upsert to postgrest
+    :param incident: dict from oracle db
+    :return: record dict
+    """
+    record = {}
+    published_date = arrow.get(incident['CURR_DATE']).replace(tzinfo="US/Central")
+    status_date = arrow.now().format()
+    record["traffic_report_id"] = generate_record_id(incident['CALL_NUMBER'], published_date.timestamp)
+    record["published_date"] = published_date.format()
+    record["traffic_report_status"] = "ACTIVE"
+    record["traffic_report_status_date_time"] = status_date
+    record["address"] = incident['ADDRESS'].strip()
+    record["issue_reported"] = incident['DESCRIPTION'].strip()
+    record["latitude"] = incident['LATITUDE']
+    record["longitude"] = incident['LONGITUDE']
+    return record
+
+
+def parse_records(traffic_incidents):
+    """
+    :param traffic_incidents: list of traffic incidents from oracle db
+    :return: list of formatted incident records
+    """
+    records = []
+    for incident in traffic_incidents:
+        record = format_record(incident)
+        records.append(record)
+
+    return records
+
+
+def apply_archive_status(records):
+    """
+    updates status to ARCHIVED and status_date_time to now
+    :param records: list of records that are in postgrest but no longer in oracle db (no longer active)
+    :return: list of records (dicts)
+    """
+    for record in records:
+        record["traffic_report_status"] = "ARCHIVED"
+        record["traffic_report_status_date_time"] = arrow.now().format()
```

Comment:
I think we should be defining the timezone here in case it's running on some other machine?

Reply:
good call!